src/App.tsx

```diff
@@ -12,6 +12,12 @@ export class App extends Component<{}, State> {
     movies: [],
   };
 
+  addMovie = (stateMovie: Movie) => {
+    this.setState(state => ({
+      movies: [...state.movies.filter(movie => movie.imdbID !== stateMovie.imdbID), stateMovie],
+    }));
```

Comment:
check the existence of the film in list before adding movie to state

```suggestion
      movies: [...state.movies, newMovie],
    }));
```